components/script/dom/document.rs

```diff
@@ -2784,9 +2790,57 @@ impl Document {
             shadow_roots: DomRefCell::new(HashSet::new()),
             shadow_roots_styles_changed: Cell::new(false),
             media_controls: DomRefCell::new(HashMap::new()),
+            csp_list: DomRefCell::new(None),
         }
     }
 
+    pub fn set_csp_list(&self, csp_list: Option<CspList>) {
+        *self.csp_list.borrow_mut() = csp_list;
+    }
+
+    pub fn get_csp_list(&self) -> Option<Ref<CspList>> {
+        ref_filter_map(self.csp_list.borrow(), Option::as_ref)
+    }
+
+    /// https://www.w3.org/TR/CSP/#should-block-request
+    pub fn should_request_be_blocked_by_csp(&self, request: &RequestBuilder) -> csp::CheckResult {
```

Comment:
Aren't those methods also defined in the net crate? Why are they duplicated?

Reply:
Different signature. [This one](hhttps://github.com/servo/servo/pull/24315/commits/9de86647524baa18d69cc4f448d3d8aeb3af127e#diff-30a18e04d7e0b66aafdf192e416cad44R2806) takes a RequestBuilder, [this one](https://github.com/servo/servo/pull/24315/commits/9de86647524baa18d69cc4f448d3d8aeb3af127e#diff-e8cd2dd2c9989c693e87a030a1755bccR143) takes a Request.

Reply:
Actually, thinking about it, I'm not sure if I actually need this any more. Doing it here means I can avoid doing IPC for a rejected request, but we probably really don't need to optimize the rejected request pipeline.